### ISSUE TITLE
When does pgBackRest need to delete or update from the repo?

### ISSUE THREAD

**Issue by Xof** [pgbackrest/pgbackrest#1148]
This is more a feature inquiry than a specific issue. We would like to use pgBackRest for archival storage, using an IAM role that is write-only (basically, granting PutObject only). Would that work properly with pgBackRest, setting aside (obviously) retention policy issues?

**Comment by sfrost** [comment]
pgbackrest needs GetObject() for various things like grabbing the prior manifest, required for incremental or differential backups to work, and then does work with archive.info, backup.info, and the .copy versions of those, to figure out which backups have completed and which are in-progress or not yet done, and is also needed for doing verification that the WAL being archived actually matches the database ID in the WAL...

Basically, no, this won't work properly and supporting it would end up throwing away a bunch of pgbackrest features and safety checks...

**Comment by dwsteele** [comment]
In principal pgBackRest is mostly write-only (as opposed to overwriting), but there are some exceptions and the most notable is cleaning up a backup path during resume or removing an incomplete backup when resume is disabled. GetObject is definitely require for many operations and removing that dependency would be a big deal, but I don't see why reading is a problem for archival storage.

It might work to disable DeleteObject in the archive path but for backups you'd need disable DeleteObject after it was completed. And disable expiration, of course. Before S3 we had a special directory we created backups in before moving them into the final location, but that doesn't work with S3 so we had to change the method.

I do like the idea in principal, but there seem to be a few practical issues. Our intention is to overwrite as little as possible, and in future repo versions we may do no overwrites, but deletes seem necessary for cleaning up backups in progress at the very least. Thoughts?

**Comment by Xof** [comment]
Having GetObject still work would be reasonable. The use-case here is limit the damage that compromising the IAM credentials pgBackRest uses creates. If DeleteObject isn't available, the attacker can get the backup, but at least can't destroy it.

**Comment by dwsteele** [comment]
This may be a use case for object versioning and/or object locking.
